CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/GalleryFragment.kt

```diff
@@ -122,17 +123,19 @@ class GalleryFragment internal constructor() : Fragment() {
 
                 // Launch the intent letting the user choose which app to share with
                 startActivity(Intent.createChooser(intent, getString(R.string.share_hint)))
-            }
+
+            } ?: Toast.makeText(activity, R.string.share_error, Toast.LENGTH_SHORT).show()
         }
 
         // Handle delete button press
         view.findViewById<ImageButton>(R.id.delete_button).setOnClickListener {
-            AlertDialog.Builder(view.context, android.R.style.Theme_Material_Dialog)
-                    .setTitle(getString(R.string.delete_title))
-                    .setMessage(getString(R.string.delete_dialog))
-                    .setIcon(android.R.drawable.ic_dialog_alert)
-                    .setPositiveButton(android.R.string.yes) { _, _ ->
-                        mediaList.getOrNull(mediaViewPager.currentItem)?.let { mediaFile ->
+            mediaList.getOrNull(mediaViewPager.currentItem)?.let { mediaFile ->
+
+                AlertDialog.Builder(view.context, android.R.style.Theme_Material_Dialog)
+                        .setTitle(getString(R.string.delete_title))
+                        .setMessage(getString(R.string.delete_dialog))
+                        .setIcon(android.R.drawable.ic_dialog_alert)
+                        .setPositiveButton(android.R.string.yes) { _, _ ->
```

Comment:
Sir the way I do it is better way to work in Kotlin. Yes the output would be same for both.

Reply:
Sorry, I meant #133. You are submitting duplicate changes in two PRs. Please remove these changes from here, so we can review them in #133 separately

Reply:
Sir I resolved both of the issues that's why both come in same PR

Reply:
As, you already closed previous one that's #133 so, could you please do check both of it in this PR

Reply:
This is being addressed in #118. If you want to keep these changes here, then please see [my suggestion](https://github.com/android/camera-samples/pull/118#issuecomment-571213362) for a better approach (disabling the button instead of checking the state after the button is clicked).

Reply:
Two of the suggestions are addressed in latest push